production/scrapers/wisconsin_population.R

```diff
@@ -0,0 +1,203 @@
+        "WRC", 
+        "CONTRACT FACILITIES", 
+        "SUBTOTAL-MALES ALL LOCATIONS", 
+        "MAXIMUM SECURITY INST", 
+        "MEDIUM SECURITY INST", 
+        "MEDIUM SECURITY INST", 
+        "CENTER SYSTEM", 
+        "CONTRACT BEDS", 
+        "FEMALE", 
+        "MALE", 
+        "SUB-TOTAL FEMALES ALL LOCATIONS", 
+        "MINIMUM SECURITY INST", 
+        "DIVISION OF JUVENILE CORRECTIONS", 
+        "TOTAL ON-GROUNDS POPULATION 3", 
+        "SUBTOTAL-MALES", 
+        "SUBTOTAL-FEMALES")
+    
+    exp_names <- c("x", "design_capacity", "total_population", "dai", "dcc")
+    clean_names <- c("Name", "Capacity.Drop", "Residents.Population", "DAI.Drop", "DCC.Drop")
+    
+    if (length(x) == 4) {
+        x1 <- wi_pop_clean_cols(x[[1]])
+        check_names(x1, exp_names)
+        names(x1) <- clean_names
+        
+        x2 <- wi_pop_clean_cols(x[[2]])
+        check_names(x2, exp_names)
+        names(x2) <- clean_names
+        
+        x3 <- wi_pop_clean_cols(x[[3]]) 
+        check_names(x3, exp_names)
+        names(x3) <- clean_names
+        
+        x4 <- wi_pop_clean_cols(x[[4]])
+        check_names(x4, c("x", "total_capacity", "total_supervised_population"))
+        names(x4) <- c("Name", "Capacity.Drop", "Residents.Population")
+    }
+    
+    # ExtractTable occasionally reads 5 tables instead of 4 
+    if (length(x) == 5){
+        x1 <- wi_pop_clean_cols(x[[1]])
+        check_names(x1, exp_names)
+        names(x1) <- clean_names
+        
+        x2 <- wi_pop_clean_cols(x[[2]])
+        check_names(x2, exp_names)
+        names(x2) <- clean_names
+        
+        x3 <- wi_pop_clean_cols(x[[3]]) 
+        check_names(x3, exp_names)
+        names(x3) <- clean_names
+        
+        x_ <- x[[4]]
+        if (x_[1,1] == "WRC (DDES FACILITY)"){
+            names(x_) <- clean_names
+            x2 <- bind_rows(x2, x_)
+        }
+        
+        x4 <- wi_pop_clean_cols(x[[5]])
+        check_names(x4, c("x", "total_capacity", "total_supervised_population"))
+        names(x4) <- c("Name", "Capacity.Drop", "Residents.Population")
+    }
+    
+    out <- bind_rows(x1, x2, x3, x4) %>% 
+        as_tibble() %>%
+        select(Name, Residents.Population) %>% 
+        mutate(Residents.Population = as.numeric(gsub("[^0-9]", "", Residents.Population)), 
+               Name = clean_fac_col_txt(Name, to_upper = TRUE)) %>% 
+        filter(!Name %in% drop_names) %>%
+        
+        # Consolidate rows to match COVID reporting aggregation 
+        mutate(Name = case_when(
+            Name %in% c("RACINE", "STURTEVANT TRANSITIONAL FACILITY") ~ 
+                "RACINE CORRECTIONAL INSTITUTION STURTEVANT TRANSITIONAL FACILITY",
+            Name %in% c("LINCOLN HILLS SCHOOL", "COPPER LAKE SCHOOL") ~ 
+                "COPPER LAKE SCHOOL LINCOLN HILLS SCHOOL", 
+            TRUE ~ Name)) %>%
+        group_by(Name) %>% 
+        summarise(Residents.Population = sum(Residents.Population)) %>% 
```

Comment:
In addition to what you've done below, you might consider adding in another line here like 
`summarise(n_facs = n())`. This is just another check that the sum of aggregated rows isn't above what you would expect it to be, and could add a line like `assertthat::see_if(n < 3)` or something

Reply:
That's a super good call. Latest commit goes one step further and throws a warning if any facility names are summed besides the 3 that are expected (love to be extra verbose with warnings). Better safe than sorry!